server/src/main/java/org/opensearch/gateway/BaseShardResponse.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Base response class for shard response.
+ *
+ * @opensearch.internal
+ */
+public abstract class BaseShardResponse {
```

Comment:
@amkhar why do we need this class?

Reply:
Removing it, as it'll be pushed as part of   https://github.com/opensearch-project/OpenSearch/pull/12504/files#diff-f22c9cb55b850db5c383ce5ebfa022765e7089d220e80bd216b208cc0e7864fcR315-R318